modules/src/main/java/org/archive/modules/extractor/ExtractorSitemap.java

```diff
@@ -181,4 +197,32 @@ private void recordOutlink(CrawlURI curi, URL newUri, Date lastModified,
 
     }
 
+    public String getUrlPattern() {
+        return urlPattern;
+    }
+
+    /**
+     * If urlPattern is not null then any url marked as a sitemap and matching the pattern is
+     * assumed to be a sitemap. Otherwise the mime-type is checked (must be "text/xml" or "application/xml") and the
+     * file is "sniffed" for the expected start of a sitemap file.
+     * @param urlPattern the url pattern to match
+     */
+    public void setUrlPattern(String urlPattern) {
+        this.urlPattern = urlPattern;

```

Comment:
Maybe we should compile the regular expression here (`Pattern.compile(urlPattern)`) or just take a Pattern object instead of a String? Other places in the config like MatchesRegexDecideRule accept a Pattern object. That way if someone makes a syntax error in the pattern it'll be caught immediately when the job is built instead of later in the crawl when they first encounter a sitemap.